tiledb/bioimg/converters/base.py

```diff
@@ -316,24 +375,46 @@ def _convert_level_to_tiledb(
     attr_dtype = reader.level_dtype(level)
     schema = get_schema(dim_names, dim_shape, max_tiles, attr_dtype, compressor)
 
+    channel_index = source_axes.dims.find("C")
+    channel_count = source_shape[channel_index] if channel_index > -1 else 1
+    channel_min_max = [
+        [
+            np.iinfo(reader.level_dtype(level)).max,
+            np.iinfo(reader.level_dtype(level)).min,
+        ]
+        for _ in range(channel_count)
+    ]
+
     # get or create TileDB array uri
     uri, created = rw_group.get_or_create(f"l_{level}.tdb", schema)
     if created:
         # write image and metadata to TileDB array
         with open_bioimg(uri, "w") as out_array:
             out_array.meta.update(reader.level_metadata(level), level=level)
+            min_max_axes = tuple(
+                idx for idx in range(len(source_axes.dims)) if idx != channel_index
+            )
+
             if chunked or max_workers:
                 inv_axes_mapper = axes_mapper.inverse
+                min_max_lock = Lock()
 
-                def tile_to_tiledb(level_tile: Tuple[slice, ...]) -> None:
+                def tile_to_tiledb(level_tile: Tuple[slice, ...], lock: Lock) -> None:
                     source_tile = inv_axes_mapper.map_tile(level_tile)
                     image = reader.level_image(level, source_tile)
                     out_array[level_tile] = axes_mapper.map_array(image)
 
+                    with lock:
+                        compute_channel_minmax(image, min_max_axes, channel_min_max)
+
```

Comment:
Have you thought avoiding `locking`. I am always inclined in avoiding locking? Did you try to calculate the min-max values after the ingestion? You can even parallelise on that using a different split scheme for the images instead of `iter_tiles`.